### PR TITLE
refactor code, fix some minor issues

### PR DESCRIPTION
**Pull request by GLEF1X**
_Please check the following:_

-   [X] Do the tests still pass?[^1]
-   [X] Does the code comply with the style guide?
    -   [X] Run `make lint` from the Wagtail root.

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

**Comment by lb-** [comment]
@GLEF1X thanks but I recommend you raise a smaller set of PRs with focusing on bug fixes. 
If you find a bug - and fix it - that's awesome but we may need a bit more info in the PR with what's being fixed and how to reproduce. Sometimes it may be very obvious but best to lean towards more communication in the PR. 

As for other minor refractors - it may be harder to merge these in without some kind of context, justification and maybe earlier discussion on an issue. 

For example - lots of changes here appear to remove the declaration of a response variable in tests. What's the problem here and why should this be changed? If this is a mere convention, maybe it doesn't matter really which way we do this, if it could cause confusion or future tests to have issues, best to raise a new enhancement suggestion issue with describing the problem. 

Also, when it comes to small 'preferences' on how to write some code, we should try to discuss and also work out how we can ensure developers can easily align with the preferred approach in the future. This may be documentation, linting or an explicit decision that some things do not really matter. 

We do appreciate the contributions but just need a bit more context to help move forward with them.

File: wagtail/contrib/settings/tests/generic/test_model.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def test_request_or_site_with_request_result_caching(self):
 
                 # only the first lookup should result in a query
                 with self.assertNumQueries(1):
-                    for i in range(4):
+                    for _ in range(4):
```
Comment (gasman) [inline wagtail/contrib/settings/tests/generic/test_model.py:53]: We generally avoid using `_` as a placeholder variable, as it interferes with the common Django pattern of using it as an alias for `gettext` / `gettext_lazy`.

Comment (GLEF1X) [reply wagtail/contrib/settings/tests/generic/test_model.py:53]: @gasman, thank you for your response! You are right; I will revert these changes. Although, I believe that in `wagtail/test/benchmark.py` it's not gonna interfere with Django translation aliases.

**Comment by GLEF1X** [comment]
@lb-, thank you for your feedback and recommendations! I will take your advice! What about response variables I believe they are unused and only confuse developers. When I looked at the tests and these response variables, I expected them to be used later in the test, but they weren't. It's confusing for me why they should be kept in code if it's clear that the test client that sends HTTP requests gets a response.

Other changes that make literal declarations instead of calls to `.append` or `__setitem__` is helpful because it adds a performance boost(it's miserable, but still, why not?), and doesn't the code quality doesn't change at all.

In [wagtail/contrib/settings/views.py](https://github.com/wagtail/wagtail/pull/9889/files#diff-1b8511bbd47182a3cbc4099e5d8426e648c02b2e3b8b3f5d9d1cec6ac61d4a73) and [wagtail/search/views/queries.py](https://github.com/wagtail/wagtail/pull/9889/files#diff-6785dcba31a9bbae6bc1e5f8883e4550e3aa07e5805b011067e9e8d63e554c2e) I removed unused variables declaration.

I'm only unsure about the [wagtail/admin/tests/pages/test_edit_page.py](https://github.com/wagtail/wagtail/pull/9889/files#diff-af279db108ccf16fa62b926938e314919e27b6df8919fa2dc07a7124f17dc9bd). The question is if it's intended to create `go_live_at` and don't pass it to `post_data`?

File: wagtail/search/backends/elasticsearch5.py [hunk 2]
```diff
@@ -577,10 +577,9 @@ def get_content_type_filter(self):
         return {"match": {"content_type": content_type}}
 
     def get_filters(self):
-        filters = []
+        filters = [self.get_content_type_filter()]
 
         # Filter by content type
```
Comment (gasman) [inline wagtail/search/backends/elasticsearch5.py:582]: This comment should be moved up.

File: wagtail/test/benchmark.py [hunk 3]
```diff
@@ -12,7 +12,7 @@ def test(self):
         memory_usage = []
         tracemalloc.start()
 
-        for i in range(self.repeat):
+        for _ in range(self.repeat):
```
Comment (gasman) [inline wagtail/test/benchmark.py:15]: Please revert

**Review by gasman — CHANGES_REQUESTED** [review]

**Review by gasman — APPROVED** [review]
Thanks @GLEF1X!

**Comment by gasman** [comment]
Merged in 768d8f3d2c100db91c20d158cc7ba6644d68fff1

**Pull request closed without merging**